chatsky/core/message.py

```diff
@@ -290,9 +282,9 @@ class level variables to store message information.
             ]
         ]
     ] = None
-    annotations: Optional[Dict[str, Any]] = None
-    misc: Optional[Dict[str, Any]] = None
-    original_message: Optional[Any] = None
+    annotations: Optional[Dict[str, Union[BaseModel, JsonValue]]] = None
+    misc: Optional[Dict[str, Union[BaseModel, JsonValue]]] = None
```

Comment:
Change type annotation to allow deeper BaseModel usage (e.g. a dictionary or a list with BaseModel values).

Reply:
```python
PydanticValue: TypeAlias = Union[
    List["PydanticValue"],
    Dict[str, "PydanticValue"],
    BaseModel,
    str,
    bool,
    int,
    float,
    None,
]
```